lite2/client.go

```diff
@@ -862,3 +906,20 @@ func (c *Client) Update(now time.Time) error {
 
 	return nil
 }
+
+func (c *Client) previousTrustedHeight(height int64) (int64, error) {
+	FirstTrustedHeight, err := c.FirstTrustedHeight()
+	if err != nil {
+		return -1, err
+	}
+	if height < FirstTrustedHeight {
```

Comment:
we won't need to check this if we do (immediately after we should check if it's not expired I think. before checking if we can trust it)